deploy/nexodus-monitoring/base/prometheus-service.yaml

```diff
@@ -2,13 +2,11 @@ apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: prometheus
-  namespace: observability
 ---
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
```

Comment:
Are we going to be able to create cluster roles in the rosa cluster?

Reply:
I think we will run into the following:
<img width="1327" alt="image" src="https://user-images.githubusercontent.com/103255/234262927-a333c8b1-a70f-44ad-be07-c2d54252730d.png">

Reply:
Not sure. I've moved this into the `dev` overlay for now.
Not sure whether we'll need service accounts, RBAC and all that goodness for ROSA.

Reply:
Update: Looks like OpenShift only needs the `ServiceMonitor` - assuming we use OpenShift Metrics - so we'd just keep that in the `base` and move the `Prometheus` object over to the `dev` overlay.
I'll hold off doing that in this PR though as there are probably lots of other things we'll find out in discussion with the folks running the ROSA cluster - i.e they may want to install the prom operator (or may have already done so for another project).